test/Constraints/diagnostics.swift

```diff
@@ -953,7 +953,7 @@ func SR_6272_a() {
     case bar
   }
 
-  // expected-error@+2 {{binary operator '*' cannot be applied to operands of type 'Int' and 'Float'}} {{35-35=Int(}} {{42-42=)}}
+  // expected-error@+2 {{binary operator '*' cannot be applied to operands of type 'Int' and 'Float'}} {{35-35=Int(}} {{43-43=)}}
```

Comment:
Why did this diagnostic change?

Reply:
It was previously using an incorrect source location, it was inserting the right paren of the cast before the existing right paren, not after. This was fixed by not removing the outer CoerceExpr when we convert a constructor call to a literal initialization.

Reply:
Then it should be in a separate PR. Tag @rudkx or @xedin for review.